src/services/cognitiveServicesAccount/format.ts

```diff
@@ -0,0 +1,232 @@
+import { ThrottlingRule } from '@azure/arm-cognitiveservices'
+import cuid from 'cuid'
+import { RawAzureCognitiveServicesAccount } from './data'
+import {
+  AzureCognitiveServicesAccount,
+  AzureCognitiveServicesAccountThrottlingRule,
+} from '../../types/generated'
+import { formatTagsFromMap, transformSystemData } from '../../utils/format'
+
+const formatThrottlingRules = (
+  rules: ThrottlingRule[]
+): AzureCognitiveServicesAccountThrottlingRule[] => {
+  return (
+    rules?.map(
+      ({
+        key,
+        renewalPeriod,
+        count,
+        minCount,
+        dynamicThrottlingEnabled,
+        matchPatterns = [],
+      }) => ({
+        key,
+        renewalPeriod,
+        count,
+        minCount,
+        dynamicThrottlingEnabled,
+        matchPatterns: matchPatterns.map(({ path, method }) => ({
+          id: cuid(),
+          path,
+          method,
+        })),
+      })
+    ) || []
+  )
+}
+
+export default ({
+  service,
+  account: subscriptionId,
+}: {
+  service: RawAzureCognitiveServicesAccount
+  account: string
+}): AzureCognitiveServicesAccount => {
+  const {
+    id,
+    name,
+    type,
+    region,
+    resourceGroupId,
+    Tags = {},
+    etag,
+    kind,
+    sku,
+    identity: { type: iType, tenantId, principalId } = {},
+    systemData,
+    properties: {
+      provisioningState,
+      endpoint,
+      internalId,
+      capabilities = [],
+      isMigrated,
+      migrationToken,
+      skuChangeInfo: {
+        countOfDowngrades,
+        countOfUpgradesAfterDowngrades,
+        lastChangeDate,
+      } = {},
+      customSubDomainName,
+      networkAcls: {
+        defaultAction,
+        ipRules = [],
+        virtualNetworkRules = [],
+      } = {},
+      encryption: {
+        keyVaultProperties: {
+          keyName,
+          keyVersion,
+          keyVaultUri,
+          identityClientId,
+        } = {},
+        keySource: encrytionKeySource,
+      } = {},
+      userOwnedStorage = [],
+      privateEndpointConnections = [],
+      publicNetworkAccess,
+      apiProperties: {
+        qnaRuntimeEndpoint,
+        qnaAzureSearchEndpointKey,
+        qnaAzureSearchEndpointId,
+        statisticsEnabled,
+        eventHubConnectionString,
+        storageAccountConnectionString,
+        aadClientId,
```

Comment:
any way to build connections to this ad data?

Reply:
Yes, I believe it can be done. Let me trial some more 😄

Reply:
I added the connection logic and field, but I'm unable to deploy the needed resources to test this in our sandbox account

Reply:
ok sounds good. I suppose we can just 🤞 lol